### PR TITLE
Notification instantiation fixed

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/NEX-778
Fixed 
1. Logic of counting and instantiation for notifications 
2. PSR-12

### CODE REVIEW

File: models/classes/notification/implementation/AbstractSqlNotificationService.php [hunk 1]
```diff
@@ -198,7 +198,7 @@ public function changeStatus(Notification $notification): bool
                 $notification->getId(),
             ];
 
-        return $persistence->exec($updateQuery, $data);
+        return 0 !== $persistence->exec($updateQuery, $data);
```
Comment (mike-ionut-mihai-sandu-tao) [inline models/classes/notification/implementation/AbstractSqlNotificationService.php:201]: hey are you sure about this ? 
can we use a casting or something similar

Comment (boajer) [reply models/classes/notification/implementation/AbstractSqlNotificationService.php:201]: `::exec` returns number of affected rows by update; so I think my proposal is the most compatible to the previous solution, and this usage https://github.com/oat-sa/tao-core/blob/af4b9eea4c8aaeedc0b6d14c964fd5a7339c673d/models/classes/notification/implementation/NotificationServiceAggregator.php#L98

File: models/classes/notification/implementation/AbstractSqlNotificationService.php [hunk 2]
```diff
@@ -208,6 +208,7 @@ private function createNotification(array $notificationDetail): Notification
             $notificationDetail[self::NOTIFICATION_FIELD_TITLE],
             $notificationDetail[self::NOTIFICATION_FIELD_MESSAGE],
             $notificationDetail[self::NOTIFICATION_FIELD_SENDER],
+            $notificationDetail[self::NOTIFICATION_FIELD_SENDER_NAME],
```
Comment (gabrielfs7) [inline models/classes/notification/implementation/AbstractSqlNotificationService.php:211]: Just to know how this is related to the PR scope. It is an improvement, right?

Comment (boajer) [reply models/classes/notification/implementation/AbstractSqlNotificationService.php:211]: This is something missing at https://github.com/oat-sa/tao-core/pull/2431 (we've merged it  today:( ), crucial for the functionality.

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by mike-ionut-mihai-sandu-tao** [comment]
this can be tested only with specific notification preconditions from client config,
not testable on local

**Comment by siwane** [comment]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**